### ISSUE TITLE
Prevent `HTTP_PROXY` from being passed to workspace commands

### ISSUE THREAD

**Issue by dcarrion87** [coder/coder#5029]
Wanted to drop this here around Tailscale integration.

We set `http_proxy` and `https_proxy` environment variables on the coder workspace container to a Tailscale proxy (deployment and service) living in the same namespace to allow the coder agent to tunnel back to the coder instance under the constraints of the label applied to the Tailscale proxy machine.

[coder workspace - http_proxy] -->  [service] --> [tailscale proxy] --> [tailnet] --> [tailscale sidecar] --> [coder instance]

This works well, but some apps are performance impacted in the coder workspace having all their traffic tunnelled via the Tailscale proxy as well. We get around this by adding a bunch of endpoints for the `no_proxy` environment variable in the users workspace that need the performance.

I'm trying to think of ways so only the coder binary is using the `http_proxy` for the purposes of getting back to the coder instance but coming stuck. If I was somehow able to edit the `coder_agent.main.init_script` to set a wrapper for the `/tmp/coder.RANDOM/coder` binary that gets created to set `http_proxy` in there would that work or would connecting users get those environment variables too with `./coder agent` reading them in? We'd need the whole binary since coder overrides things like `GIT_SSH_COMMAND` that would need to have this too. I'm conscious making adjustments to that init_script would break things for us in the future after a coder update that changed it.

Any thoughts welcome on this.

[Edit]: Just tested that theory and no go. I guess I'm looking for a way to set proxy variable that coder binary will use for connectivity but not pass it down to its children.

**Comment by dcarrion87** [comment]
@kylecarbs would it be possible to solve this by adding a parameter/arguments to the coder agent to specify a proxy without it passing the proxy onto children / sessions?

**Comment by dcarrion87** [comment]
@kylecarbs this is still an issue for us. As per my PR I don't have enough knowledge of how the coder agent, server and tailscale stuff works to understand if it is even going to make a different at all parts.

**Comment by kylecarbs** [comment]
@dcarrion87 could you specify `HTTP_PROXY` and `HTTPS_PROXY` in your Coder agent block to override them to empty values? That should prevent them from being passed to children processes.

**Comment by dcarrion87** [comment]
@kylecarbs sorry I forgot to update this. We did get around this by running an `unset HTTP_PROXY` and `unset HTTPS_PROXY` in the /etc/ global startup scripts to block users from getting it on login and overriding the Coder variables to use a wrapper in /usr/local/bin that injected these in.

Since then we've decided to just completely axe the proxy and implement segregated routing capabilities out the Kubernetes cluster. It's just too much of a headache maintaining these proxies and we need to ensure network performance due to the nature of the workload and someone accidentally getting the proxy causes issues. We would have needed to use that proxy for other things and it's just not fast enough.
